### PR TITLE
use maps.MergeCopy

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/model/virtualservice.go [hunk 1]
```diff
@@ -380,38 +381,20 @@ func mergeHTTPMatchRequest(root, delegate *networking.HTTPMatchRequest) *network
...
+	out.Headers = maps.MergeCopy(root.Headers, delegate.Headers)
+
 	// withoutheaders
-	if len(root.WithoutHeaders) > 0 && len(out.WithoutHeaders) == 0 {
-		out.WithoutHeaders = make(map[string]*networking.StringMatch)
-	}
-	for k, v := range root.WithoutHeaders {
-		out.WithoutHeaders[k] = v
-	}
+	maps.MergeCopy(root.WithoutHeaders, delegate.WithoutHeaders)
```
Comment (ramaraochavali) [inline pilot/pkg/model/virtualservice.go:387]: need to copy to out?

Comment (ramaraochavali) [reply pilot/pkg/model/virtualservice.go:387]: same at other places

Comment (hzxuzhonghu) [reply pilot/pkg/model/virtualservice.go:387]: my mistake

File: pkg/maps/maps.go [hunk 2]
```diff
@@ -40,12 +40,20 @@ func Keys[M ~map[K]V, K comparable, V any](m M) []K {
 	return maps.Keys(m)
 }
 
+// Copy copies all key/value pairs in src adding them to dst.
+// When a key in src is already present in dst,
+// the value in dst will be overwritten by the value associated
+// with the key in src.
+func Copy[M1 ~map[K]V, M2 ~map[K]V, K comparable, V any](dst M1, src M2) {
```
[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pkg/maps/maps.go:47]: I used it before

**Review by ramaraochavali — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: pilot/pkg/model/virtualservice.go [hunk 3]
```diff
@@ -380,38 +381,20 @@ func mergeHTTPMatchRequest(root, delegate *networking.HTTPMatchRequest) *network
...
 	}
 
 	// SourceLabels
-	if len(root.SourceLabels) > 0 && len(out.SourceLabels) == 0 {
-		out.SourceLabels = make(map[string]string)
-	}
-	for k, v := range root.SourceLabels {
-		out.SourceLabels[k] = v
-	}
+	out.SourceLabels = maps.MergeCopy(root.SourceLabels, delegate.SourceLabels)
```
Comment (howardjohn) [inline pilot/pkg/model/virtualservice.go:397]: is this intended to change behavior? I don't see where delegate.SourceLabels was ever added before

Comment (hzxuzhonghu) [reply pilot/pkg/model/virtualservice.go:397]: Yes it is.  Because out=delegate, so that is automatically

**Pull request merged**